newrelic/resource_newrelic_workflow.go

```diff
@@ -0,0 +1,371 @@
+						"type": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: fmt.Sprintf("(Required) The type of the destination. One of: (%s).", strings.Join(listValidWorkflowsDestinationTypes(), ", ")),
+						},
+					},
+				},
+			},
+			"issues_filter": {
+				Type:        schema.TypeSet,
+				Required:    true,
+				Description: "Workflow's destination configuration.",
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						// Required
+						"name": {
+							Type:        schema.TypeString,
+							Required:    true,
+							Description: "(Required) Filter's name.",
+						},
+						"type": {
+							Type:         schema.TypeString,
+							Required:     true,
+							ValidateFunc: validation.StringInSlice(listValidWorkflowsFilterTypes(), false),
+							Description:  fmt.Sprintf("(Required) The type of the filter. One of: (%s).", strings.Join(listValidWorkflowsFilterTypes(), ", ")),
+						},
+
+						// Optional
+						"predicates": {
+							Type:        schema.TypeList,
+							Optional:    true,
+							Description: "Workflow's destination configuration.",
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"attribute": {
+										Type:        schema.TypeString,
+										Required:    true,
+										Description: "(Required) predicate's attribute.",
+									},
+									"operator": {
+										Type:         schema.TypeString,
+										Required:     true,
+										ValidateFunc: validation.StringInSlice(listValidWorkflowsOperatorTypes(), false),
+										Description:  fmt.Sprintf("The type of the operator. One of: (%s).", strings.Join(listValidWorkflowsOperatorTypes(), ", ")),
+									},
+									"values": {
+										Type:        schema.TypeList,
+										Required:    true,
+										Description: "List of predicate values.",
+										Elem:        &schema.Schema{Type: schema.TypeString},
+									},
+								},
+							},
+						},
+
+						// Computed
+						"filter_id": {
+							Type:        schema.TypeString,
+							Computed:    true,
+							Description: "filter id.",
+						},
+					},
+				},
+			},
+			"muting_rules_handling": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringInSlice(listValidMutingRulesTypes(), false),
+				Description:  fmt.Sprintf("The type of the muting rule handling. One of: (%s).", strings.Join(listValidMutingRulesTypes(), ", ")),
+			},
+
+			// Optional
+			"workflow_enabled": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Indicates whether the workflow is enabled.",
+			},
+			"enrichments_enabled": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Indicates whether the enrichments are enabled.",
+			},
+			"destinations_enabled": {
+				Type:        schema.TypeBool,
+				Optional:    true,
+				Description: "Indicates whether the destinations are enabled.",
+			},
+			"enrichments": {
```

Comment:
this is a block definition for enrichments, right?  so shouldn't this be called `enrichment` because each block represents a single enrichment?